src/channel.c

```diff
@@ -53,6 +53,29 @@
 #include "packet.h"
 #include "session.h"
 
+/*
+ * libssh2_channel_read_host_info
+ *
+ * Pass shost, sport, and shost_len from library to user space application.
+ *
+ * Return 0 on success, LIBSSH2_ERROR_BAD_USE if channel/shost is not initialized.
+ */
+LIBSSH2_API int
+libssh2_channel_read_host_info(LIBSSH2_CHANNEL *channel, unsigned char *shost,
+                               unsigned int *sport, unsigned int *shost_len)
+{
+    int rc = 0;
+
+    if (!channel || !channel->shost)
+        return LIBSSH2_ERROR_BAD_USE;
+
+    memcpy(shost, channel->shost, strlen((char*)channel->shost));
```

Comment:
this seems like asking for an overflow. I would prefer the API to pass in a buffer size too to make that impossible. Possibly then return a failure if the buffer was too small.

Reply:
I have updated to add a buffer_size for the user to provide and a check to make sure it is big enough.